exercise/views.py

```diff
@@ -379,9 +379,24 @@ def get_common_objects(self):
                     'name': name,
                     'content': response.text,
                     'html': 'text/html' in response.headers.get('Content-Type'),
+                    'data': response.content,
                 })
         self.note('models')
 
+    def get(self, request, *args, **kwargs):
+        if request.GET.get("download", False):
+            try:
+                bytedata = self.models[0]['data']
+            except OSError:
+                return HttpResponseNotFound()
+
+            response = HttpResponse(bytedata,
+                content_type="application/octet-stream")
```

Comment:
It doesn't look like it is necessary to send this response as byte data when the text (string type) is also available. `HttpResponse` could just use the text data, i.e., the original `response.text` from `get_common_objects()`.

https://docs.djangoproject.com/en/3.2/ref/request-response/#passing-strings

It looks like the old code has never really handled binary (non-text) files. Rendering binary files as text in the html page outputs garbled nonsensical content. It would be better to avoid that, but that can be another task in the future.